### PR TITLE
Add GCS Version 2 integration using Google cloud API

### PR DESCRIPTION
**Pull request by LuQQiu**
Add GCS Version 2 integration using Google cloud API

### CODE REVIEW

**Comment by LuQQiu** [comment]
This PR has already tested with nested mount + bin/alluxio runTests 
and 
```
bin/alluxio runUfsTests --path gs://<bucket> -Dalluxio.underfs.gcs.version=2 -Ds.gcs.credential.path=<credential>
```

**Comment by LuQQiu** [comment]
@bf8086 PTAL， thanks!

File: docs/en/ufs/GCS.md [hunk 1]
```diff
@@ -48,9 +62,11 @@ specify an **existing** GCS bucket and directory as the under storage system by
 alluxio.master.mount.table.root.ufs=gs://GCS_BUCKET/GCS_DIRECTORY
 ```
 
-The google credentials must also be specified for the root mount point. In
-`conf/alluxio-site.properties`, add:
+Choose your desired GCS implementation way and provide the corresponding Google credentials.
```
Comment (bf8086) [inline docs/en/ufs/GCS.md:65]:
```suggestion
Choose your preferred GCS UFS version and provide the corresponding Google credentials.
```

File: docs/en/ufs/GCS.md [hunk 2]
```diff
@@ -26,6 +26,20 @@ the bucket, or using an existing one. For the purposes of this guide, the GCS bu
 For more information on GCS, please read its
 [documentation](https://cloud.google.com/storage/docs/overview).
 
+## Setup 
+
+Alluxio has two GCS integrations. The default GCS UFS module (GCS version 1) is implemented based on
```
Comment (bf8086) [inline docs/en/ufs/GCS.md:31]:
```suggestion
Alluxio provides two ways to access GCS. The default GCS UFS module (GCS version 1) is implemented based on 
```

File: docs/en/ufs/GCS.md [hunk 3]
```diff
@@ -26,6 +26,20 @@ the bucket, or using an existing one. For the purposes of this guide, the GCS bu
...
 [documentation](https://cloud.google.com/storage/docs/overview).
 
+## Setup 
+
+Alluxio has two GCS integrations. The default GCS UFS module (GCS version 1) is implemented based on 
+[jets3t](http://www.jets3t.org/) library which is design for AWS S3. 
+Thus, it only accepts Google cloud storage interoperability access/secret keypair 
+which allows full access to all Google cloud storages inside a Google cloud project.
+No permission or access control can be placed on the interoperability keys.
+The conjuction of Google interoperability API and jets3t library also impact the performance of default GCS UFS module.
```
Comment (bf8086) [inline docs/en/ufs/GCS.md:36]:
```suggestion
The conjunction of Google interoperability API and jets3t library also impact the performance of the default GCS UFS module. 
```

**Review by bf8086 — CHANGES_REQUESTED** [review]
I took a look at mostly the documentation assuming we have already reviewed the code previously.

**Comment by LuQQiu** [comment]
@bf8086 Applied all your suggestions, PTAL, thanks!

**Review by bf8086 — APPROVED** [review]
LGTM. Thanks!

**Comment by LuQQiu** [comment]
alluxio-bot, sync-merge this please

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

**Pull request merged**